examples/auctions/simple_open_auction.v.py

```diff
@@ -0,0 +1,76 @@
+# Open Auction
+
+# Auction params
+# Beneficiary recieves money from the highest bidder
+beneficiary: public(address)
+auction_start: public(timestamp)
+bidding_time: public(num(sec))
+
+# Current state of auction
+highest_bidder: public(address)
+highest_bid: public(num(wei))
+
+# Allowed withdrawals of previous bids
+pending_returns:(num(wei)[address])
+
+# Set to true at the end, disallows any change
+ended: public(bool)
+
+# Create a simple auction with `_bidding_time`
+# seconds bidding time on behalf of the
+# beneficiary address `_beneficiary`.
+def __init__(_beneficiary: address, _bidding_time: num(sec)):
+    self.beneficiary = _beneficiary
+    self.bidding_time = _bidding_time
+    self.auction_start = block.timestamp
+
+# Bid on the auction with the value sent
+# together with this transaction.
+# The value will only be refunded if the
+# auction is not won.
+@payable
+def bid():
+    # Check if bidding period is over.
+    assert block.timestamp < (self.auction_start)+ self.bidding_time
+    # Check if bid is high enough
+    assert msg.value > self.highest_bid
+    if not self.highest_bid == 0:
+        # Allows the previous highest bidder to take their money back
+        self.pending_returns[self.highest_bidder] += self.highest_bid
+    self.highest_bidder = msg.sender
+    self.highest_bid = msg.value
+
+# Withdraw a bid that was overbid
+def withdraw():
```

Comment:
In conjunction with comment about auction completion and refunding money, perhaps this function can have an argument of the person to refund with a default of `msg.sender` or something similar so that the same function to refund money automatically can be used by a person during the auction to get their money back.